pkg/manager/http.go

```diff
@@ -734,6 +738,42 @@ func (serv *HTTPServer) modulesInfo(w http.ResponseWriter, r *http.Request) {
 	serv.jsonPage(w, r, "modules", cover.Modules)
 }
 
+func (serv *HTTPServer) httpAddInput(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "multipart/form-data")
+	if r.Method != http.MethodPost {
+		http.Error(w, "only POST method supported", http.StatusMethodNotAllowed)
+		return
+	}
+	err := r.ParseMultipartForm(20 << 20)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to parse form: %v", err), http.StatusBadRequest)
+		return
+	}
+	file, _, err := r.FormFile("file")
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to retrieve file from form-data: %v", err), http.StatusBadRequest)
+		return
+	}
+	defer file.Close()
+	data, err := io.ReadAll(file)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to read file: %v", err), http.StatusBadRequest)
+		return
+	}
+	p, err := serv.Cfg.Target.Deserialize(data, prog.NonStrict)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to deserialize prog: %v", err), http.StatusBadRequest)
+		return
+	}
+	data = p.Serialize()
+	sig := hash.String(data)
+	if serv.CorpusDB != nil {
+		serv.CorpusDB.Save(sig, data, 0)
```

Comment:
Do you want to add the program directly to the corpus or rarther let it be executed on the VMs and go though our full pkg/fuzzer pipeline?

In that case it's better to push it via `pkg/fuzzer`'s `AddCandidates`.

Reply:
FWIW here's how I did this in my local patch:

```
func (mgr *Manager) injectProg(w http.ResponseWriter, r *http.Request) {
        r.ParseForm()
        progToInject := strings.TrimSpace(r.PostFormValue("injected_prog"))
        progToInject = strings.ReplaceAll(progToInject, "\r", "") 
        fmt.Printf("progToInject: %q\n", progToInject)
        data := UIInjectedProg{}
        if progToInject != "" {
                data.Prog = strings.Split(progToInject, "\n")
                byteProg := []byte(progToInject)
                p, err := mgr.target.Deserialize(byteProg, prog.NonStrict)
                if err == nil {
                        fmt.Printf("p: %v\n", p)
                        candidate := rpctype.Candidate{
                                Prog:      byteProg,
                                Minimized: true,
                                Smashed:   true,
                        }   
                        candidates := []rpctype.Candidate{candidate}
                        mgr.addNewCandidates(candidates)
                } else {
                        fmt.Printf("failed to deserialize: %v\n", err)
                }   
        }   
        executeTemplate(w, injectProgTemplate, data)
}
```

Reply:
updated this PR instead